src/main/scala/com/databricks/spark/xml/XmlRelation.scala

```diff
@@ -60,13 +61,43 @@ case class XmlRelation protected[spark] (
     }
   }
 
-  override def buildScan: RDD[Row] = {
+  override def buildScan(): RDD[Row] = {
     StaxXmlParser.parse(
       baseRDD(),
       schema,
       parseConf)
   }
 
+  override def buildScan(requiredColumns: Array[String]): RDD[Row] = {
+    val requiredFields = requiredColumns.map(schema(_))
+    val schemaFields = schema.fields
+    if (schemaFields.deep == requiredFields.deep) {
+      buildScan()
+    } else if (parseConf.failFastFlag) {
+      val safeRequestedSchema = StructType(requiredFields)
+      StaxXmlParser.parse(
+        baseRDD(),
+        safeRequestedSchema,
+        parseConf)
+    } else {
+      // If `parseConf` is disabled, then it needs to parse all the values
+      // so that we can decide which row is malformed.
+      val safeRequestedSchema = StructType(
+        requiredFields ++ schema.fields.filterNot(requiredFields.contains(_)))
+      val rows = StaxXmlParser.parse(
+        baseRDD(),
+        safeRequestedSchema,
+        parseConf)
+
+      val rowSize = requiredFields.length
+      rows.mapPartitions { iter =>
```

Comment:
I am not too sure if it is okay to call `mapPartition` again to slice. I could do this in `StaxXmlParser.parse()` but I decided that this way looks much easy to read.

However, as this logic runs at most executions (by default, this logic would be executed), I am still puzzled if it is okay.